src/pages/settings/PasswordPage.js

```diff
@@ -1,25 +1,158 @@
+                                />
+                            </View>
+                            <View style={styles.mb6}>
+                                <Text style={[styles.mb1, styles.formLabel]}>New Password*</Text>
+                                <TextInput
+                                    secureTextEntry
+                                    autoCompleteType="password"
+                                    textContentType="password"
+                                    style={styles.textInput}
+                                    value={this.state.newPassword}
+                                    onChangeText={newPassword => this.setState({newPassword})}
+                                />
+                            </View>
+                            <View style={styles.mb6}>
+                                <Text style={[styles.mb1, styles.formLabel]}>Confirm New Password*</Text>
+                                <TextInput
+                                    secureTextEntry
+                                    autoCompleteType="password"
+                                    textContentType="password"
+                                    style={styles.textInput}
+                                    value={this.state.confirmNewPassword}
+                                    onChangeText={confirmNewPassword => this.setState({confirmNewPassword})}
+                                />
+                            </View>
+                            {this.props.account.requiresTwoFactorAuth ? (
+                                <View style={styles.mb6}>
+                                    <Text style={[styles.mb1, styles.formLabel]}>Two Factor Code</Text>
+                                    <TextInput
+                                        style={styles.textInput}
+                                        value={this.state.twoFactorCode}
+                                        onChangeText={twoFactorCode => this.setState({twoFactorCode})}
+                                        placeholder="Required when 2FA is enabled"
+                                    />
+                                </View>
+                            ) : null}
+                            {!isEmpty(this.props.account.error) && (
+                                <Text style={styles.formError}>
+                                    {this.props.account.error}
+                                </Text>
+                            )}
+                        </View>
+                        <View style={styles.fixedBottomButton}>
+                            <ButtonWithLoader
+                                isDisabled={!this.state.currentPassword || !this.state.newPassword
+                                    || !this.state.confirmNewPassword
+                                    || (this.state.newPassword !== this.state.confirmNewPassword)
+                                    || (this.props.account.requiresTwoFactorAuth && !this.state.twoFactorCode)}
```

Comment:
Idea (not a blocker): also disable if `newPassword === currrentPassword`, so that you can't try to change your password to the same one.